src/databricks/labs/ucx/assessment/assessment.py

```diff
@@ -60,20 +87,229 @@ def _verify_ws_client(w: WorkspaceClient):
+
+    def _mock_loader(self):
+        return None
+
+    def _try_fetch(self) -> list[Table]:
+        for row in self._fetch(
+            f'SELECT * FROM {self._schema}.{self._table}'
+        ):
+            yield Table(*row)
+
+
+class ExternalLocationCrawler(CrawlerBase):
+
+    def __init__(self, sbe: SqlBackend, ws: WorkspaceClient, catalog, schema):
+        super().__init__(sbe, catalog, schema, "external_locations")
+        self._ws = ws
+
+    def _external_locations(self, tables: [ExtLoc]):
+        mounts = MountsCrawler(self._backend, self._ws, self._schema).snapshot()
+        ext_locations = []
+        for table in tables:
+            location = table.location
+            if location is not None and len(location) > 0:
+                if location.startswith("dbfs:/mnt"):
+                    for mount in mounts:
+                        if location[5:0].startswith(mount.name):
+                            location = location.replace(mount.name,mount.source)
+                            break
+                if not location.startswith("dbfs"):
+                    dupe = False
+                    loc = 0
+                    while loc < len(ext_locations) and not dupe:
+                        common = os.path.commonprefix([ext_locations[loc].location, os.path.dirname(location) + '/'])
+                        if common.count("/") > 2:
+                            ext_locations[loc] = ExtLoc(common)
+                            dupe = True
+                        loc += 1
+                    if not dupe:
+                        ext_locations.append(ExtLoc(os.path.dirname(location) + '/'))
+        return ext_locations
+
+    def _ext_loc_list(self):
+        crawler = InventoryTableCrawler(self._backend, self._catalog, self._schema)
+        table_list = crawler.snapshot()
+        return self._external_locations(table_list)
+
+    def snapshot(self) -> list[ExtLoc]:
+        return self._snapshot(self._try_fetch, self._ext_loc_list)
+
+    def _try_fetch(self) -> list[ExtLoc]:
+        for row in self._fetch(
+            f'SELECT * FROM {self._schema}.{self._table}'
+        ):
+            yield ExtLoc(*row)
+
+
+class MountsCrawler(CrawlerBase):
+    def __init__(self, backend: SqlBackend, ws: WorkspaceClient, inventory_database: str):
+        super().__init__(backend, "hive_metastore", inventory_database, "mounts")
+        self._dbutils = ws.dbutils
+
+    def inventorize_mounts(self):
+        self._append_records(self._list_mounts())
+
+    def _list_mounts(self):
+        mounts = []
+        for mount_point, source, _ in self._dbutils.fs.mounts():
+            mounts.append(Mount(mount_point, source))
+        return mounts
+
+    def snapshot(self) -> list[Mount]:
+        return self._snapshot(self._try_fetch, self._list_mounts)
+
+    def _try_fetch(self) -> list[Mount]:
+        for row in self._fetch(
+            f'SELECT * FROM {self._schema}.{self._table}'
+        ):
+            yield Mount(*row)
+
```

Comment:
```suggestion
```

we already have support for mounts: https://github.com/databrickslabs/ucx/commit/436223754398727e54310216eb917fb4f4298864

Reply:
Added snapshot method to the original class.